Controller/LanguageSwitcherController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * File containing the LanguageSwitcherController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\DemoBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\Core\MVC\Symfony\Routing\RouteReference;
+use Symfony\Component\HttpFoundation\Request;
+
+class LanguageSwitcherController extends Controller
+{
+    /**
+     * @param Request $request
+     * @param RouteReference $routeReference
+     *
+     * Displays the language switcher
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showSwitcherAction( Request $request, RouteReference $routeReference )
+    {
+        /** @var \eZ\Publish\Core\Helper\TranslationHelper $translationHelper */
+        $translationHelper = $translationHelper = $this->container->get( 'ezpublish.translation_helper' );
+        $siteaccess = [];
+        $availableLanguages = [];
+
+        // create an array for corresponding siteaccesses names depending on the lang
+        foreach ( $translationHelper->getAvailableLanguages() as $lang )
+        {
+            if ( $lang != null )
```

Comment:
If you have only language installed SA (for example eng) and all the default setting after running the wizard, `$translationHelper->getAvailableLanguages()` will return:

```
array (size=2)
  0 => null
  1 => string 'eng-GB' (length=6)
```